### PR TITLE
fix: maxlength of vendorfield

### PR DESCRIPTION
**Pull request by kalgoop**
## Proposed Change
take substring of value in vendorfield, if larger than 64 characters

if value is already less than 64 characters, substring function returns the same string

### CODE REVIEW

**Comment by alexbarnsley** [comment]
It would be better to throw an error I think. taking a substring could result in undesired effects for the person doing it

**Comment by faustbrian** [comment]
As @alexbarnsley said an error is the way to do it as this could result in an order not being able to be paid.

File: lib/index.js [hunk 1]
```diff
@@ -72,6 +72,8 @@ module.exports = class ArkPay extends Emittery {
      */
     vendorField (value) {
       this.data.transfer.vendorField = value
+      
+      if (value.length > 64) this.emit('error', 'Value in venderfield is too long')
```
Comment (faustbrian) [inline lib/index.js]: The emitter is not yet used at this point. It should simply throw an error.

Comment (faustbrian) [reply lib/index.js]: Also the wording is odd, should be `The given value exceeds the maximum length of 64 characters.`.

**Review by faustbrian — APPROVED** [review]

**Pull request merged**